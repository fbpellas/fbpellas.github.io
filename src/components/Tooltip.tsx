import * as React from 'react';
import { OverlayTrigger, Tooltip as TooltipBootstrap } from 'react-bootstrap';

export interface TooltipProps {
  word: string;
  definition: string;
  className?: string;
}

const Tooltip: React.FC<TooltipProps> = (props) => {
  const { className = 'text-inline bottom-dot', definition, word } = props;

  return (
    <OverlayTrigger
      key="bottom"
      placement="bottom"
      overlay={<TooltipBootstrap id={`tooltip-bottom`}>{definition}</TooltipBootstrap>}
    >
      <div className={className}>{word}</div>
    </OverlayTrigger>
  );
};

export { Tooltip };
