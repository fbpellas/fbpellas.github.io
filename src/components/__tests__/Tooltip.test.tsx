import React from 'react';
import { render } from '@testing-library/react';
import { Tooltip } from '../Tooltip';

test('Tooltip', () => {
  const { container, rerender } = render(<Tooltip definition='def' word='word' />);
  expect(container).toMatchSnapshot('className default')

  rerender(<Tooltip className='something' definition='def' word='word' />)
  expect(container).toMatchSnapshot('className provided');
});