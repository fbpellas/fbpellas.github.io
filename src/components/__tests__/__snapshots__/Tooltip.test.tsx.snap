// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip: className default 1`] = `
<div>
  <div
    class="text-inline bottom-dot"
  >
    word
  </div>
</div>
`;

exports[`Tooltip: className provided 1`] = `
<div>
  <div
    class="something"
  >
    word
  </div>
</div>
`;
